scipy/stats/_crosstab.py

```diff
@@ -0,0 +1,154 @@
+        does not occur in the corresponding sequence in `levels`, that value
+        is ignored and not counted in the returned array `count`.
+
+    Returns
+    -------
+    elements : tuple of numpy.ndarrays.
+        Tuple of length ``len(args)`` containing the arrays of elements that
+        are counted in `count`.  These can be interpreted as the labels of
+        the corresponding dimensions of `count`.
+    count : numpy.ndarray
+        Counts of the unique elements in ``zip(*args)``, stored in an array.
+        Also known as a *contingency table* when ``len(args) > 1``.
+
+    See Also
+    --------
+    numpy.unique
+
+    Notes
+    -----
+    .. versionadded:: 1.5.0
+
+    References
+    ----------
+    .. [1] "Contingency table", http://en.wikipedia.org/wiki/Contingency_table
+
+    Examples
+    --------
+    >>> from scipy.stats import crosstab
+
+    Given the lists `a` and `x`, create a contingency table that counts the
+    frequencies of the corresponding pairs.
+
+    >>> a = ['A', 'B', 'A', 'A', 'B', 'B', 'A', 'A', 'B', 'B']
+    >>> x = ['X', 'X', 'X', 'Y', 'Z', 'Z', 'Y', 'Y', 'Z', 'Z']
+    >>> (avals, xvals), count = crosstab(a, x)
+    >>> avals.tolist()
+    ['A', 'B']
+    >>> xvals.tolist()
+    ['X', 'Y', 'Z']
+    >>> count
+    array([[2, 3, 0],
+           [1, 0, 4]])
+
+    So `('A', 'X')` occurs twice, `('A', 'Y')` occurs three times, etc.
+
+    Higher dimensional contingency tables can be created.
+
+    >>> p = [0, 0, 0, 0, 1, 1, 1, 0, 0, 1]
+    >>> (avals, xvals, pvals), count = crosstab(a, x, p)
+    >>> count
+    array([[[2, 0],
+            [2, 1],
+            [0, 0]],
+           [[1, 0],
+            [0, 0],
+            [1, 3]]])
+    >>> count.shape
+    (2, 3, 2)
+
+    The values to be counted can be set by using the `levels` argument.
+    It allows the elements of interest in each input sequence to be
+    given explicitly instead finding the unique elements of the sequence.
+
+    For example, suppose one of the arguments is an array containing the
+    answers to a survey question, with integer values 1 to 4.  Even if the
+    value 1 does not occur in the data, we want an entry for it in the table.
+
+    >>> q1 = [2, 3, 3, 2, 4, 4, 2, 3, 4, 4, 4, 3, 3, 3, 4]  # 1 does not occur.
+    >>> q2 = [4, 4, 2, 2, 2, 4, 1, 1, 2, 2, 4, 2, 2, 2, 4]  # 3 does not occur.
+    >>> options = [1, 2, 3, 4]
+    >>> vals, count = crosstab(q1, q2, levels=(options, options))
+    >>> count
+    array([[0, 0, 0, 0],
+           [1, 1, 0, 1],
+           [1, 4, 0, 1],
+           [0, 3, 0, 3]])
+
+    If `levels` is given, but an element of `levels` is None, the unique values
+    of the corresponding argument are used. For example,
+
+    >>> vals, count = crosstab(q1, q2, levels=(None, options))
+    >>> vals
+    [array([2, 3, 4]), [1, 2, 3, 4]]
+    >>> count
+    array([[1, 1, 0, 1],
+           [1, 4, 0, 1],
+           [0, 3, 0, 3]])
+    """
```

Comment:
?

```
L0 = len(args[0])
if L0 == 0:
```

etc?

Reply:
@matthew-brett, I don't understand this comment/question.  Could you elaborate?

Reply:
I just meant that, rather than recalculating `len(args[0])` multiple times, it will be very slightly quicker, and perhaps neater, to calculate once and put into a variable.  Very minor point.  It's just that whenever I see recalculation, my Spidey-sense tingles - so I have to suppress it with the argument to myself that it's so quick it probably doesn't matter.

Reply:
Gotcha, thanks.  I use both `len(args)` and `len(args[0])` in multiple places, so I'll define variables for those.

Reply:
Fixed in the update.